### PR TITLE
Adjust scope of locals declared within Expression Statements according to the latest design.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #12597.

@dotnet/roslyn-compiler, @gafter Please review.

### CODE REVIEW

**Comment by jcouv** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs [hunk 1]
```diff
@@ -123,12 +123,13 @@ internal Binder Binder
 
         public static SourceLocalSymbol MakeOutVariable(
             Symbol containingSymbol,
-            Binder binder,
+            Binder scopeBinder,
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs]: Some internal documentation about these would be very helpful, please.

**Comment by gafter** [comment]
Other than the one comment, :shipit:

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs]: Sure, in one of the follow-up PRs.

**Pull request merged**